explainability-connectors/src/main/java/org/kie/trustyai/connectors/utils/ListenableFutureUtils.java

```diff
@@ -0,0 +1,47 @@
+package org.kie.trustyai.connectors.utils;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ForkJoinPool;
+
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+
+/**
+ * Convert a {@link ListenableFuture} into a {@link CompletableFuture}
+ *
+ * @param <T>
+ */
+public class ListenableFutureUtils<T> {
+
+    private final CompletableFuture<T> completableFuture;
+
+    public ListenableFutureUtils(ListenableFuture<T> listenableFuture) {
+        this.completableFuture = new CompletableFuture<T>() {
+            @Override
+            public boolean cancel(boolean mayInterruptIfRunning) {
+                boolean cancelled = listenableFuture.cancel(mayInterruptIfRunning);
+                super.cancel(cancelled);
+                return cancelled;
```

Comment:
I'm not a fan of "extreme single lining" but isn't this equivalent to
```suggestion
                return super.cancel(listenableFuture.cancel(mayInterruptIfRunning));
```
?